File: Recursion/recursion.c
// Recusion Program in C Language to Find the factorial of a number

#include <stdio.h>
#include <conio.h>

int factorial(int n);

int main(void) {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    int result = factorial(num);
    printf("Factorial of %d = %d\n", num, result);

    return 0;
}

int factorial(int n) {

    if (n == 0) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

File: Recursion/recursion.rb
# Program to implement Recursion in Ruby

def factorial(n)
if n == 0
    return 1
else 
    return n * factorial(n-1)
end
end

puts "Enter a number: "
input = gets.chomp  # take raw input

# Check if input is a valid integer (regex ensures only digits allowed)
if input.match?((/\A\d+\z/))
   num = input.to_i
   
   result = factorial(num)
   puts "Factorial of number #{num} is #{result}"
else
    puts "Enter a valid Non negative integer value"
end



